content/blog/kubernetes-is-better-with-pulumi/index.md

```diff
@@ -0,0 +1,94 @@
+
+```ts
+var frontendServiceType string
+if isMinikube {
+   frontendServiceType = “ClusterIP”
+} else {
+   frontendServiceType = “LoadBalancer”
+}
+
+frontendService, err := corev1.NewService(ctx, “frontent”, &corev1.ServiceArgs{
+   Metadata: &metav1.ObjectMetaArgs{
+      Labels: frontendLabels,
+      Name: pulumi.String(“frontend”),
+   },
+   Spec: &corev1.ServiceSpecArgs{
+      Type: pulumi.String(frontendServiceType),
+      Ports: corev1.ServicePortArray{
+         corev1.ServicePortArg{
+            Port: pulumi.Int(80),
+         },
+      },
+      Selector: frontendLabels,
+   },
+})
+```
+
+- **Strongly typed resources** – A tricky part about defining Kubernetes resources in YAML/JSON is the lack of typing. It's hard to track typos, and you have to apply the resource to a cluster to get any feedback. Pulumi helps here by showing type information right in the SDK.
+
+![Typed resources](typed_resources.jpg)
+
+- **Readiness logic** – Pulumi includes readiness logic to help you understand when your Kubernetes resources are ready. This makes it easy to integrate Kubernetes workflows into CI systems! Unlike `kubectl apply`, Pulumi doesn’t indicate success until all of the resources are ready to go. You won’t have to use `kubectl get …” repeated to see if resources are available.
+
+![Readiness logig](rediness_logig.gif)
+
+- **Deprecation warnings** – Pulumi warns you if you’re using Kubernetes resources with deprecated apiVersions, and gives you info about which versions remove them entirely. Don’t be caught by surprise when you upgrade your clusters!
+
+![Deprecation warnings](deprecation_warnings.jpg)
+
+## For  Operators
+
+Pulumi is not just for developers, operators have fine-grained control over Kubernetes that includes managing clusters, reuse of existing YAML manifests and Helm charts, and a convenience API for Kubernetes that abstracts boilerplate such as networking configurations. You can even export your Pulumi programs to YAML for portability.
+
+- **Manage Kubernetes clusters** – With Pulumi, you can manage every layer of your stack, from specifying a custom VPC to deploying Kubernetes clusters! [Check out this example]({{< relref "/blog/multicloud-app" >}}) where we create clusters on AWS, AKS, and GKE, and then deploy an app to all of them.
+
+- **Use existing YAML manifests** – Many of us already have a pile of YAML Kubernetes definitions to manage. Manage these manifests directly; no conversion needed! Pulumi shows detailed previews and allows you to run arbitrary transformations on the YAML before it is applied.
+
+![YAML manifests](yaml_manifests.jpg)
```

Comment:
Might want to make this into a Python snippet rather than using the image here.

Reply:
Python snippet added